From 1f3f72e8e10111b08fe5b024116c952f5c1153aa Mon Sep 17 00:00:00 2001
Message-Id: <1f3f72e8e10111b08fe5b024116c952f5c1153aa.1653582481.git.git@grubix.eu>
From: Michael J Gruber <git@grubix.eu>
Date: Thu, 26 May 2022 18:18:17 +0200
Subject: [PATCH] test: make T450 independent of application/octet-stream
 interpretation

The actual content type of `application/octet-stream` is up to content
type detection of the reader, and thus may not be stable across
implementations or versions. This showed up when

fd46fc19 ("emacs:  document/defcustom notmuch-multipart/alternative-discouraged", 2022-05-14)

introduced a test for omitting a part of type `text/html` because it
expected a part of type `application/octet-stream` to remain in place,
i.e. a part of "unstable type". In particular, tests with `fd46fc19`
would succeed on RHEL/EPEL but fail on all current Fedoras with

```
| FAIL   multipart/alternative hides html by default
|	--- T450-emacs-show.16.notmuch-show-multipart-alternative	2022-05-26 15:34:42.100557244 +0000
|	+++ T450-emacs-show.16.OUTPUT	2022-05-26 15:34:42.102557207 +0000
|	@@ -24,7 +24,7 @@
|	 uses 64 as the
|	 buffer size.
|	 [ text/html (hidden) ]
|	-[ 0001-Deal-with-situation-where-sysconf-_SC_GETPW_R_SIZE_M.patch: application/octet-stream (as text/x-diff) ]
|	+[ 0001-Deal-with-situation-where-sysconf-_SC_GETPW_R_SIZE_M.patch: application/octet-stream (as text/x-patch) ]
|	 From e3bc4bbd7b9d0d086816ab5f8f2d6ffea1dd3ea4 Mon Sep 17 00:00:00 2001
|	 From: Alexander Botero-Lowry <alex.boterolowry@gmail.com>
|	 Date: Tue, 17 Nov 2009 11:30:39 -0800
```

due to the different type detected.

Fix this by giving that message a specicific type of `text/x-diff` in
the test corpus, and adjust all affected test outputs.

Signed-off-by: Michael J Gruber <git@grubix.eu>
---
 test/corpora/default/bar/baz/05:2,                              | 2 +-
 .../notmuch-show-multipart-alternative                          | 2 +-
 test/emacs.expected-output/raw-message-cf0c4d-52ad0a            | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/test/corpora/default/bar/baz/05:2, b/test/corpora/default/bar/baz/05:2,
index 75b05fa4..02b23965 100644
--- a/test/corpora/default/bar/baz/05:2,
+++ b/test/corpora/default/bar/baz/05:2,
@@ -63,7 +63,7 @@ and <a href=3D"http://mail-index.netbsd.org/pkgsrc-bugs/2006/06/07/msg01680=
 
 --0016e687869333b14e0478963d33--
 --0016e687869333b1570478963d35
-Content-Type: application/octet-stream; 
+Content-Type: text/x-diff; 
 	name="0001-Deal-with-situation-where-sysconf-_SC_GETPW_R_SIZE_M.patch"
 Content-Disposition: attachment; 
 	filename="0001-Deal-with-situation-where-sysconf-_SC_GETPW_R_SIZE_M.patch"
diff --git a/test/emacs-show.expected-output/notmuch-show-multipart-alternative b/test/emacs-show.expected-output/notmuch-show-multipart-alternative
index e44926bc..e2951d2b 100644
--- a/test/emacs-show.expected-output/notmuch-show-multipart-alternative
+++ b/test/emacs-show.expected-output/notmuch-show-multipart-alternative
@@ -24,7 +24,7 @@ http://mail-index.netbsd.org/pkgsrc-bugs/2006/06/07/msg016808.htmlspecifically
 uses 64 as the
 buffer size.
 [ text/html (hidden) ]
-[ 0001-Deal-with-situation-where-sysconf-_SC_GETPW_R_SIZE_M.patch: application/octet-stream (as text/x-diff) ]
+[ 0001-Deal-with-situation-where-sysconf-_SC_GETPW_R_SIZE_M.patch: text/x-diff ]
 From e3bc4bbd7b9d0d086816ab5f8f2d6ffea1dd3ea4 Mon Sep 17 00:00:00 2001
 From: Alexander Botero-Lowry <alex.boterolowry@gmail.com>
 Date: Tue, 17 Nov 2009 11:30:39 -0800
diff --git a/test/emacs.expected-output/raw-message-cf0c4d-52ad0a b/test/emacs.expected-output/raw-message-cf0c4d-52ad0a
index 75b05fa4..02b23965 100644
--- a/test/emacs.expected-output/raw-message-cf0c4d-52ad0a
+++ b/test/emacs.expected-output/raw-message-cf0c4d-52ad0a
@@ -63,7 +63,7 @@ and <a href=3D"http://mail-index.netbsd.org/pkgsrc-bugs/2006/06/07/msg01680=
 
 --0016e687869333b14e0478963d33--
 --0016e687869333b1570478963d35
-Content-Type: application/octet-stream; 
+Content-Type: text/x-diff; 
 	name="0001-Deal-with-situation-where-sysconf-_SC_GETPW_R_SIZE_M.patch"
 Content-Disposition: attachment; 
 	filename="0001-Deal-with-situation-where-sysconf-_SC_GETPW_R_SIZE_M.patch"
-- 
2.36.1.258.gea251483f2

